qiskit/circuit/library/probability_distributions/normal.py

```diff
@@ -0,0 +1,254 @@
+            mu: The parameter :math:`\mu`, which is the expected value of the distribution.
+                Can be either a float for a 1d random variable or a list of floats for a higher
+                dimensional random variable. Defaults to 0.
+            sigma: The parameter :math:`\sigma^2`, which is the variance or covariance
+                matrix. Default to the identity matrix of appropriate size.
+            bounds: The truncation bounds of the distribution as tuples. For multiple dimensions,
+                ``bounds`` is a list of tuples ``[(low0, high0), (low1, high1), ...]``.
+                If ``None``, the bounds are set to ``(-1, 1)`` for each dimension.
+            upto_phase: If True, load the probabilities up to a global phase for a more efficient
+                circuit.
+            name: The name of the circuit.
+        """
+        _check_dimensions_match(num_qubits, mu, sigma, bounds)
+        _check_bounds_valid(bounds)
+
+        # set default arguments
+        dim = 1 if isinstance(num_qubits, int) else len(num_qubits)
+        if mu is None:
+            mu = 0 if dim == 1 else [0] * dim
+
+        if sigma is None:
+            sigma = 1 if dim == 1 else np.eye(dim)
+
+        if bounds is None:
+            bounds = (-1, 1) if dim == 1 else [(-1, 1)] * dim
+
+        if not isinstance(num_qubits, list):  # univariate case
+            super().__init__(num_qubits, name=name)
+
+            x = np.linspace(bounds[0], bounds[1], num=2**num_qubits)
+        else:  # multivariate case
+            super().__init__(sum(num_qubits), name=name)
+
+            # compute the evaluation points using numpy's meshgrid
+            # indexing 'ij' yields the "column-based" indexing
+            meshgrid = np.meshgrid(*[np.linspace(bound[0], bound[1], num=2**num_qubits[i])
+                                     for i, bound in enumerate(bounds)], indexing='ij')
+            # flatten into a list of points
+            x = list(zip(*[grid.flatten() for grid in meshgrid]))
+
+        # compute the normalized, truncated probabilities
+        probabilities = multivariate_normal.pdf(x, mu, sigma)
+        normalized_probabilities = probabilities / np.sum(probabilities)
+
+        # store the values, probabilities and bounds to make them user accessible
+        self._values = x
+        self._probabilities = normalized_probabilities
+        self._bounds = bounds
+
+        # use default the isometry (or initialize w/o resets) algorithm to construct the circuit
+        # pylint: disable=no-member
+        if upto_phase:
+            self.isometry(np.sqrt(normalized_probabilities), self.qubits, None)
+        else:
+            from qiskit.extensions import Initialize  # pylint: disable=cyclic-import
+            initialize = Initialize(np.sqrt(normalized_probabilities))
+            circuit = initialize.gates_to_uncompute().inverse()
+            self.compose(circuit, inplace=True)
```

Comment:
this relies on the current implementation detail that isometry is phase correct and initialize is only up to global phase. This could change at any point, so you shouldn't rely on it. Correcting phase should be trivial though, you can just set `circuit.global_phase`. There's no extra cost to it (the docstring says `upto_phase: If True, load the probabilities up to a global phase for a more efficient circuit.`)

Reply:
also i don't understand why you care about the global phase. you're loading a probability distribution, the phase does not alter that (this goes back to my overall comment that I don't understand why there's only one "correct" way to represent a probability distribution).

Reply:
So I checked and `isometry` is not just up to global phase but up to multiplication with a diagonal matrix. In principle, this still shouldn't matter to applications such as amplitude estimation. But after testing it, for some reason, the Grover operator is wrong if `isometry` is used instead of `initialize`. We tried to figure out why with @stefan-woerner but couldn't get to the bottom of it.
I'll open an issue discussing this in future, for now I'd rename `upto_phase` to `upto_diag` and set it to False per default (i.e. by default it yields the correct Grover operator).